cmd/transport/base-controller.go

```diff
@@ -0,0 +1,156 @@
+
+	"k8s.io/apiserver/pkg/server/mux"
+	"k8s.io/apiserver/pkg/server/routes"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/component-base/metrics/legacyregistry"
+	"k8s.io/klog/v2"
+	utilflag "k8s.io/kubernetes/pkg/util/flag"
+
+	clientopts "github.com/kubestellar/kubestellar/pkg/client-options"
+	edgeclientset "github.com/kubestellar/kubestellar/pkg/client/clientset/versioned"
+	edgeinformers "github.com/kubestellar/kubestellar/pkg/client/informers/externalversions"
+	"github.com/kubestellar/kubestellar/pkg/kbuser"
+	"github.com/kubestellar/kubestellar/pkg/transport"
+	spaceclient "github.com/kubestellar/kubestellar/space-framework/pkg/msclientlib"
+	spacemanager "github.com/kubestellar/kubestellar/space-framework/pkg/space-manager"
+)
+
+const (
+	transportControllerName         = "transport-controller"
+	defaultResyncPeriod             = time.Duration(0)
+	defaultConcurrency              = 4
+	defaultMetricsServerBindAddress = ":10472"
+	defaultKubestellarCoreSpaceName = "espw"
+	defaultTransportSpaceName       = "transport"
+	defaultSpaceProviderObjectName  = "default"
+	defaultExternalAccess           = false
+)
+
+func parseArguments(fs *pflag.FlagSet) (concurrency int, metricsServerBindAddress string, kcsName string, transportSpaceName string,
+	externalAccess bool, spaceProvider string, spaceMgtOpts *clientopts.ClientOpts) {
+	metricsServerBindAddress = defaultMetricsServerBindAddress
+	klog.InitFlags(flag.CommandLine)
+	fs.AddGoFlagSet(flag.CommandLine)
+	fs.IntVar(&concurrency, "concurrency", defaultConcurrency, "number of concurrent workers to run in parallel")
+	fs.Var(&utilflag.IPPortVar{Val: &metricsServerBindAddress}, "metrics-server-bind-address", "The IP address with port at which to serve /metrics and /debug/pprof/")
+	fs.StringVar(&kcsName, "core-space", defaultKubestellarCoreSpaceName, "the name of the KubeStellar core space")
+	fs.StringVar(&transportSpaceName, "transport-space", defaultTransportSpaceName, "the name of the transport space")
+	fs.BoolVar(&externalAccess, "external-access", defaultExternalAccess, "the access to the spaces. True when the space-provider is hosted in a space while the controller is running outside of that space")
+	fs.StringVar(&spaceProvider, "space-provider", defaultSpaceProviderObjectName, "the name of the KubeStellar space provider")
+	spaceMgtOpts = clientopts.NewClientOpts("space-mgt", "access to the space reference space")
+	spaceMgtOpts.AddFlags(fs)
+
+	fs.Parse(os.Args[1:])
+
+	return
+}
+
+func startMetricsServer(serverBindAddress string, logger logr.Logger) {
+	mymux := mux.NewPathRecorderMux(transportControllerName)
+	mymux.Handle("/metrics", legacyregistry.Handler())
+	routes.Profiling{}.Install(mymux)
+	go func() {
+		err := http.ListenAndServe(serverBindAddress, mymux)
+		if err != nil {
+			logger.Error(err, "Failure in web serving")
+			klog.FlushAndExit(klog.ExitFlushTimeout, 1)
+		}
+	}()
+}
+
+func Run(transportImplementation transport.Transport) {
```

Comment:
Who calls this? I think the step of actual execution should be a part of this PR. If gotta merge due to schedule, understandable.